recipes/easylzma/all/conanfile.py

```diff
@@ -37,33 +37,29 @@ def build(self):
             tools.patch(**patch)
         cmake = CMake(self)
         cmake.configure()
-        cmake.build()
+        cmake.build(target="easylzma" if self.options.shared else "easylzma_s")
 
-    def package(self):
+    @property
+    def _license_text(self):
         # Extract the License/s from the README to a file
         tmp = tools.load("source_subfolder/README")
-        license_contents = tmp[tmp.find("License",1):tmp.find("work.", 1)+5]
-        tools.save("LICENSE", license_contents)
-        # Package it
-        self.copy("LICENSE*", dst="licenses",  ignore_case=True, keep_path=False)
+        return tmp[tmp.find("License",1):tmp.find("work.", 1)+5]
 
-        # Copy static and dynamic libs
-        build_dir = os.path.join(self._source_subfolder,self.name + "-" + self.version)
-        if self.options.shared:
-            self.copy(pattern="*.dylib*", dst="lib", src=build_dir, keep_path=False, symlinks=True)
-            self.copy(pattern="*.so*", dst="lib", src=build_dir, keep_path=False, symlinks=True)
-            self.copy(pattern="*.dll", dst="bin", src=build_dir, keep_path=False)
-            self.copy(pattern="*.dll.a", dst="lib", src=build_dir, keep_path=False)
-        else:
-            self.copy(pattern="*.a", dst="lib", src=build_dir, keep_path=False)
-        self.copy(pattern="*.lib", dst="lib", src=build_dir, keep_path=False)
+    def package(self):
```

Comment:
Using this conditioned copy of files according to the value of options.shared was a suggestion of @uilianries. Why do you think is better to copy all of them regarding of the build_type?

Reply:
See the `build()` method.
It will only build the `shared` or `static` target.
Therefore, we don't need to be careful and can copy everything.